test/jdk/com/sun/net/httpserver/simpleserver/SecurityManagerTest.java

```diff
@@ -0,0 +1,204 @@
+            var request = HttpRequest.newBuilder(uri(server, "")).build();
+            var response = client.send(request, BodyHandlers.ofString());
+            assert response.statusCode() == 200;
+            assert response.body().equals(expectedBody);
+            assert response.headers().firstValue("content-type").get().equals("text/html; charset=UTF-8");
+            assert response.headers().firstValue("content-length").get().equals(expectedLength);
+            assert response.headers().firstValue("last-modified").get().equals(lastModifiedDir);
+        } finally {
+            server.stop(0);
+        }
+    }
+
+    private static void testFileGET() throws Exception {
+        var expectedBody = "some text";
+        var expectedLength = Integer.toString(expectedBody.getBytes(UTF_8).length);
+        var server = SimpleFileServer.createFileServer(LOOPBACK_ADDR, TEST_DIR, OutputLevel.VERBOSE);
+
+        server.start();
+        try {
+            var client = HttpClient.newBuilder().proxy(NO_PROXY).build();
+            var request = HttpRequest.newBuilder(uri(server, "aFile.txt")).build();
+            var response = client.send(request, BodyHandlers.ofString());
+            assert response.statusCode() == 200;
+            assert response.body().equals("some text");
+            assert response.headers().firstValue("content-type").get().equals("text/plain");
+            assert response.headers().firstValue("content-length").get().equals(expectedLength);
+            assert response.headers().firstValue("last-modified").get().equals(lastModifiedFile);
+        } finally {
+            server.stop(0);
+        }
+    }
+
+    @SuppressWarnings("removal")
+    private static void testCreateHandler(){
+        try {
+            SimpleFileServer.createFileServer(LOOPBACK_ADDR, TEST_DIR, OutputLevel.NONE);
+            throw new RuntimeException("Handler creation expected to fail");
+        } catch (AccessControlException expected) { }
+
+        try {
+            SimpleFileServer.createFileHandler(TEST_DIR);
+            throw new RuntimeException("Handler creation expected to fail");
+        } catch (AccessControlException expected) { }
+    }
+
+    static final String openHTML = """
+                <!DOCTYPE html>
+                <html>
+                <head>
+                <meta charset="utf-8"/>
+                </head>
+                <body>
+                """;
+
+    static final String closeHTML = """
+                </body>
+                </html>
+                """;
+
+    static URI uri(HttpServer server, String path) {
+        return URIBuilder.newBuilder()
+                .host("localhost")
+                .port(server.getAddress().getPort())
+                .scheme("http")
+                .path("/" + path)
+                .buildUnchecked();
+    }
+
+    static final DateTimeFormatter HTTP_DATE_FORMATTER =
+            DateTimeFormatter.ofPattern("EEE, dd MMM yyyy HH:mm:ss v");
```

Comment:
I think this and other usages of `DateTimeFormatter.ofPattern` need to be called with `Locale.US` (or something similar) as second argument, otherwise the current default locale will be used. I noticed because on my laptop the `Last-modified` header contains a german date.

Reply:
I just realized I commented on a test file, while the actual culprit is in `FileServerHandler.java`. But I guess it applies to all usages of this class and method.

Reply:
The problem I was referring to was not about printing to the console. I hadn't thought about that, I agree the default locale should be used there. I was referring to `Last-modified` HTTP headers with a non-English date value, e.g.:

    Last-modified: Di., 21 Sep. 2021 09:56:53 GMT

I think browsers will get confused by this. It probably isn't a big deal since I think the server doesn't implement conditional GETs (i.e. use the last-modified date in subsequent requests), but the HTTP spec is [quite strict ][1] about date formats.

[1]: https://datatracker.ietf.org/doc/html/rfc2616#section-3.3.1

Reply:
Thanks for clarifying. Yes the date should be in GMT and should not be localized when sent in a Last-Modified / Date header. Good catch!

Reply:
Good catch indeed, thanks! I'll update to GMT when used in the headers.